### DISCUSSION TITLE
Perform state transition by changing styles/classes

### DISCUSSION THREAD

**Discussion by alexnm** [teleporthq/teleport-code-generators#551] [Ideas]
This was already debated a bit inside the team. For now we only support state changes by rendering/not rendering elements, not be toggling classes on/off. But if we want to support animations, we should reopen the topic at some point

**Comment by alexnm** [comment]
Maybe we can looking into other animation models, not necessary related to class toggles 
Ex: hidden checkbox https://dev.to/winduptoy/a-javascript-free-frontend-2d3e

**Comment by paulbrie** [comment]
Great article! Learned a lot. We should definitely explore this path.
